custom_components/tahoma/coordinator.py

```diff
@@ -76,8 +76,16 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed("Failed to connect.") from exception
         except (ServerDisconnectedError, NotAuthenticatedException):
             self.executions = {}
-            await self.client.login()
-            self.devices = await self._get_devices()
+
+            # During the relogin, similar exceptions can be thrown.
+            try:
+                await self.client.login()
+                self.devices = await self._get_devices()
+            except BadCredentialsException as exception:
+                raise ConfigEntryAuthFailed() from exception
+            except TooManyRequestsException as exception:
```

Comment:
Added `TooManyRequestsException` as well, to have better visual errors for https://github.com/iMicknl/ha-tahoma/issues/476.